[Slipperybara] iP

Slipperybara opened the pull request:
# Sunny
> Sunny is the name of my dog -Jerry

It has all the function that is required up till week 2. Specifically, it is,
- text based
- Easy to learn
- ~~FAST~~SUPERFAST to use, 

Allyou need to do it, 
1. Download it from [here](https://github.com/Slipperybara/ip/releases/tag/A-Jar).
2. double click it
3. add your tasks
4. let it manage your tasks for you :shipit: 

and it is **FREE**

Features: 
- [x] managing tasks
- [ ] managing deadline and events(coming soon!!)
- [ ] Add delight to the experience when all tasks are complete :tada:

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Slipperybara committed:
Changed name to Sunny, added in welcome and goodbye message

Slipperybara committed:
Minor bug fix - change class name to Sunny

Slipperybara committed:
final fix for Level-0

Slipperybara committed:
Added in chat feature, bot will repeat user's input. Prompt "bye" will end the convo

Slipperybara committed:
Added list function, changed original code structure to OOP

Slipperybara committed:
More changes to OOP

Slipperybara committed:
Created tasks class and using them in Replies class

Slipperybara committed:
Parsed user message to mark and unmark tasks

Slipperybara committed:
Added subclasses to Tasks, added factory method "TaskCreator", improved chatbot replies

Slipperybara committed:
Exceptions for user's messages (Not all exceptions accounted for)

Slipperybara committed:
Added in delete function

Slipperybara committed:
Changed file path, added in one test case from input and output

Slipperybara committed:
Add header comments and fix indentations and naming

Slipperybara committed:
Save tasks to local hard disks, load data from hard disk when sunny starts up

Slipperybara committed:
Merge branch 'branch-Level-7'

Slipperybara committed:
Remove stored strings to reset chatbot

Slipperybara committed:
Store date and time using java.time instead of String

Slipperybara committed:
Merge branch 'branch-Level-8'

Slipperybara committed:
Rename replies to Ui, Create storage class and use in Ui class

Slipperybara committed:
Add Parser class

Slipperybara committed:
Update parser class,

Slipperybara committed:
Revert "Update parser class,"

This reverts commit 771c1382d89af9c31bbdb9822d2138fcb4de129f.

Slipperybara committed:
Revert "Revert "Update parser class,""

This reverts commit 901e49817279c972119bb54122ef973ccfc81241.

Slipperybara committed:
Merge branch 'branch-MoreOOP'

Slipperybara committed:
Package files to sunny

Slipperybara committed:
Merge branch 'branch-Packages'

Slipperybara committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

Slipperybara committed:
Change comment header

Slipperybara committed:
Test TaskCreator and Parser Class

Slipperybara committed:
Update the JavaDoc on Tests classes

Slipperybara committed:
Edit based on coding standard

Slipperybara committed:
Merge branch 'branch-CodingStandard'

Slipperybara committed:
Merge branches

Slipperybara committed:
Update mark when program rerun, update harddrive after every command

Slipperybara committed:
implment command as a seprate class, create subclasses extending from command

Slipperybara committed:
Implement the rest of command subclasses, and replaced the corresponding code in UI class

ZweZeya commented on src/main/java/sunny/Task.java:
```diff
@@ -0,0 +1,50 @@
+        this.isDone = false;
+    }
+
+    public void setter (boolean x) {
```
It's better to use a verb for a method name. It can be a descriptive name like `setCompletionStatus`.

ZweZeya commented on src/main/java/sunny/Ui.java:
```diff
@@ -0,0 +1,82 @@
+
+    // File path for external storage
+    String filePath = "/Users/jerryyou/ip/taskslist.txt";
+    Storage store = new Storage(filePath);
```
`store` might be an ambiguous name because it can be both a verb and a noun. Perhaps `storage` might be better?

ZweZeya commented on src/main/java/sunny/Parser.java:
```diff
@@ -0,0 +1,32 @@
+ * Breaks down user messages for other classes
+ */
+public class Parser {
+    private String firstHalf;
```
These names might be confusing because what is the half? Maybe you can use `firstWord` or something?

ZweZeya reviewed (commented):
Generally looks good other than the suggestions I have for the naming of methods and variables. :)

Slipperybara committed:
Implement checkstyle extension, fix coding standard

Slipperybara committed:
Switch from CLI to GUI

Slipperybara committed:
Merge branch 'branch-CheckStyle'

Slipperybara committed:
Merge branch 'branch-Level10'

QinHaichen12 commented on src/main/java/sunny/Command.java:
```diff
@@ -0,0 +1,17 @@
+     * Builds string, read and write to hard drive, depends on the subclass
+     * @param ls
+     */
+    public abstract String runCommand(List<Task> ls, String m);
```
Should this method be named to sound less like a subclass?

QinHaichen12 commented on src/main/java/sunny/DeleteCommand.java:
```diff
@@ -0,0 +1,20 @@
+    private String s = "";
+    @Override
+    public String runCommand(List<Task> ls, String m) {
+        int i2 = Integer.parseInt(m);
```
Maybe consider using a different variable name other than i2?

QinHaichen12 commented on src/main/java/sunny/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
Why did you choose to declare db as var

QinHaichen12 commented on src/main/java/sunny/EventTask.java:
```diff
@@ -0,0 +1,47 @@
+        super(description);
+    }
+
+    private String m1 = super.description.split("/from", 2)[0];
```
Maybe this variable can be renamed to something more descriptive?

QinHaichen12 reviewed (commented):
LGTM! The method names and class names are well defined and clear.

Slipperybara committed:
Connect Main class straight away to Ui class, delete Sunny class

Slipperybara committed:
Implement minor improvements following coding standard

Slipperybara committed:
Add dependencies to the build

As currently implementing GUI

Add dependencies so user can access the GUI when operating from the JAR file

Slipperybara committed:
Create assertions in Ui

Validate assumptions made in the Ui class, more specifically, the 'getResponse' method

Include assertions for when user input null message, and when task is not created by the TaskCreator class

Slipperybara committed:
Add assertions to TaskCreator class

Slipperybara committed:
Include welcome message during initialization

Introduce the name of the bot, and prompt user to input text

Slipperybara committed:
Move nest code in Ui to a new command class

Slipperybara committed:
Move nested code in 'write' method to a new method

Slipperybara committed:
Add a shortcut method to mass add tasks when reading

Slipperybara committed:
Add more javadoc on Task subclasses

Slipperybara committed:
Merge branch 'branch-CodingStandard'

Slipperybara committed:
Merge pull request #2 from Slipperybara/branch-A-CodeQuality

Branch a code quality

Slipperybara committed:
Change made to merge with 'CodeQuality'

Slipperybara committed:
Merge branch 'master' of https://github.com/Slipperybara/ip

# Conflicts:
#	src/main/java/sunny/Ui.java

Slipperybara committed:
Add assertion feature to avoid merge conflict with branch-A-Assertions

Slipperybara committed:
Merge branch 'master' into branch-A-Assertions

Slipperybara committed:
Merge pull request #1 from Slipperybara/branch-A-Assertions

Branch a assertions

Slipperybara committed:
Merge branch 'master' of https://github.com/Slipperybara/ip

Slipperybara committed:
Make improvements in code quality, add snooze command

Slipperybara committed:
Fix bug on writing

Status of tasks will always become undone after reloading

Use the write method after every task altering command such as 'mark', 'unmark', and 'delete

Slipperybara committed:
Change character and background image

Slipperybara committed:
Change background color to blue

Slipperybara committed:
Remove excess code in the file

Slipperybara committed:
Add exception to include invalid deadlines

Slipperybara committed:
Add exception to include invalid event timeline

Slipperybara committed:
Add javadoc for WrongMessageException class

Slipperybara committed:
Make minor coding standard improvements

Slipperybara committed:
Create user guide

Slipperybara committed:
Update user guide

Slipperybara committed:
Create tests for storage class

Slipperybara committed:
Make minor improvements

Slipperybara committed:
Make minor changes

Slipperybara committed:
Change storage structure to comply with jar

Slipperybara committed:
Update user guide

Slipperybara committed:
Include more information in exception message

Slipperybara committed:
Update user guide

Slipperybara committed:
Fix javaFX missing component issue
